controllers/object_controls.go

```diff
@@ -3663,6 +3716,48 @@ func getDaemonsetControllerRevisionHash(ctx context.Context, daemonset *appsv1.D
 	return hash, nil
 }
 
+func TransformDRADriverController(obj *appsv1.Deployment, n ClusterPolicyController) error {
+	config := n.singleton.Spec.DRADriver
+	// update image
+	image, err := gpuv1.ImagePath(&config)
+	if err != nil {
+		return err
+	}
+	obj.Spec.Template.Spec.Containers[0].Image = image
+
+	// update image pull policy
+	obj.Spec.Template.Spec.Containers[0].ImagePullPolicy = gpuv1.ImagePullPolicy(config.ImagePullPolicy)
+
+	// set image pull secrets
+	if len(config.ImagePullSecrets) > 0 {
+		addPullSecrets(&obj.Spec.Template.Spec, config.ImagePullSecrets)
+	}
+
+	return nil
+}
+
+func transformDeployment(obj *appsv1.Deployment, n ClusterPolicyController) error {
+	logger := n.logger.WithValues("Deployment", obj.Name)
+	transformations := map[string]func(*appsv1.Deployment, ClusterPolicyController) error{
+		"nvidia-dra-driver-controller": TransformDRADriverController,
+	}
+
+	t, ok := transformations[obj.Name]
+	if !ok {
+		logger.Info(fmt.Sprintf("No transformation for Deployment '%s'", obj.Name))
+		return nil
+	}
+
+	// apply per operand Deployment config
+	err := t(obj, n)
+	if err != nil {
+		logger.Error(err, "Failed to apply transformation", "resource", obj.Name)
```

Comment:
Could we drop the `resource` `obj.Name`?  We already add this when constructing the logger.